File: 2020/day_3/part_1.py
with open('input.txt') as f:
    trees_hit = 0
    x_pos = 0
    for line in f:
        line = line.rstrip('\n')
        trees_hit += int(line[x_pos] == '#')
        x_pos = (x_pos + 3) % len(line)

print(trees_hit)

File: 2023/day_01/part_1.js
const fs = require('fs');
const readline = require('readline');

let sum = 0;

function isDigit(s) {
    s = String(s);
    return s >= '0' && s <= '9';
}

const rl = readline.createInterface({
    input: fs.createReadStream('input.txt'),
});

rl.on('line', (line) => {
    let first;
    let last;
    for (const c of line) {
        if (isDigit(c)) {
            if (first == null) {
                first = c;
            }
            last = c;
        }
    }

    // console.log(line);
    if (first != null && last != null) {
        // console.log(first, last);
        const num = +(first + last);
        // console.log(num);
        sum += num;
    }
});

rl.on('close', () => {
    console.log(`Solution: ${sum}`);
});
